cep-XXXX.md

```diff
@@ -0,0 +1,153 @@
+<tr><td> Implementation </td><td> N/A </td></tr>
+</table>
+
+## Abstract
+
+This CEP aims to standardize names and other strings used to identify packages, artifacts and channels in the conda ecosystem.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT",
+"RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as
+described in [RFC2119][RFC2119] when, and only when, they appear in all capitals, as shown here.
+
+More specificlly, violations of a MUST or MUST NOT rule MUST result in an error. Violations of the
+rules specified by any of the other all-capital terms MAY result in a warning, at discretion of the
+implementation.
+
+### Identifying package artifacts
+
+The conda ecosystem distinguishes between two types of packages:
+
+- Distributable package names: represented by a concrete, downloadable, extractable conda artifact.
+- Virtual package names: not backed by any physical artifact. They only exist on the client side.
+
+#### Package names
+
+A distributable package name MUST only consist of lowercase ASCII letters, numbers, hyphens, periods and underscores. It MUST start with a letter, a number, or a single underscore. It MUST NOT include two consecutive separators (hyphen, period, underscore).
+
+Virtual package names MUST only consist of lowercase ASCII letters, numbers, hyphens, periods and underscores. They MUST NOT use two consecutive separators, with one exception: they MUST start with two underscores.
+
+This means that distributable package names MUST match the following case-insensitive regex: `^(([a-z0-9])|([a-z0-9_](?!_)))[._-]?([a-z0-9]+(\.|-|_|$))*$`.
+
+Virtual package names MUST follow this other regex: `^__[a-z0-9][._-]?([a-z0-9]+(\.|-|_|$))*$`.
+
+In all cases, the maximum length of a package name MUST NOT exceed 128 characters.
+
+#### Version strings
+
+Version strings MUST only consist of digits, periods, lowercase ASCII letters, underscores, plus
+symbols, and exclamation marks. Additional rules apply but are out of scope in this CEP and will be
+discussed separately.
+
+The maximum length of a version string MUST NOT exceed 128 characters.
+
+#### Build strings
+
+Builds strings MUST only consist of ASCII letters, numbers, periods, plus symbols, and underscores. They MUST match this regex `^[a-zA-Z0-9_\.+]+$`.
+
+The maximum length of a build string MUST NOT exceed 128 characters.
+
+#### Filenames
+
+Distributable conda artifacts MUST have a filename following this scheme:
+
+```text
+<package name>-<version string>-<build string>.<extension>
+```
+
+Virtual conda packages do not exist on disk and SHOULD NOT need filename standardization.
+
+#### Distribution strings
+
+A "distribution string" MAY be used to identify a package artifact without specifying the extension or the channel. It MUST match the following syntax:
+
+```text
+<package name>-<version string>-<build string>
```

Comment:
We need to use space-separated and/or `=`-separated forms of this as valid matchspecs in recipes. We may want to be clearer about the scope of this section or add a carve out for matchspecs

Reply:
Yea this is not a MatchSpec, is a dist str, which is not allowed anywhere, just a convention like filenames are. I'll clarify.